packages/teleport-uidl-validator/src/validator/utils.ts

```diff
@@ -251,16 +271,40 @@ Used on Node ${JSON.stringify(node)}.\nParent node is ${JSON.stringify(parent, n
   return errors
 }
 
-const dynamicPathExistsInDefinitions = (path: string, defKeys: string[]) => {
-  if (!path.includes('.')) {
-    // prop/state is a scalar value, not a dot notation
-    return defKeys.includes(path)
-  }
+const dynamicPathExistsInDefinitions = (
```

Comment:
Hello @guvvanch thanks for working on this, can we modify the function a little bit.
To support both kinds of notions like.

```js
hero.bgImage.src

hero['bgImage']['src']
```

So, we don't support just only `[]` notion while referring to props. Something like this
```js
function isValidObjectPath(path) {
  // Extract the keys from the path string
  const keys = path
    // Replace [' and '] with '.' for easier splitting
    .replace(/\['/g, '.')
    .replace(/'\]/g, '')
    // Replace . notation with . after replacing [] notation
    .replace(/\./g, ' . ')
    // Split the keys
    .split(/\[|\]|\./)
    .filter(Boolean);

  let obj = propDefinitions;

  for (const key of keys) {
    // Check if the key exists in the current object
    if (!obj[key]) {
      return false;
    }

    // Move to the next nested object
    obj = obj[key];
  }

  return true;
}

```

And add some tests too, please

Reply:
Hi @JayaKrishnaNamburu , thank you for reviewing this PR. Regarding your feedback:
1) 
```
  const keys = path
    // Replace [' and '] with '.' for easier splitting
    .replace(/\['/g, '.')
    .replace(/'\]/g, '')
    // Replace . notation with . after replacing [] notation
    .replace(/\./g, ' . ')
    // Split the keys
    .split(/\[|\]|\./)
    .filter(Boolean);
```
Upon review, it appears that this implementation has some limitations. For instance, it fails to accommodate paths with double quotes, such as `hero["bgImage"]["src"]`. Additionally, the replacement of dots `.` with spaces ` . ` (`.replace(/\./g, ' . ')`) might cause issues during key comparison due to the added spaces.

To address these issues, I've introduced a more comprehensive regular expression that handles various quotation marks and ensures more accurate key extraction:
 `const pathKeys = path.split(/\.|\[\s*['"]?(.+?)['"]?\s*\]/).filter(Boolean)`
 
 2)
I've revised the construction of the `obj` variable to account for the differences in interfaces between state/prop definitions and importDefinitions:
 
```
   let obj = Object.keys(definitions).reduce((acc, key) => {
    acc[key] =
      type === 'import'
        ? definitions[key]
        : (definitions[key] ).defaultValue

    return acc
  }, {} )
```

3)
```
  for (const key of keys) {
    // Check if the key exists in the current object
    if (!obj[key]) {
      return false;
    }

    // Move to the next nested object
    obj = obj[key];
  }
```

Here, I believe, it is better to use `if (!(key in obj)) {` instead of `  if (!obj[key]) {`, so that it doesn't return `false` when the given path exists, but its value is set to `''`, `undefined`, or `null`. 

4) These changes should be covered in test cases as I added these cases to sample project data.

Please let me know if you have any further suggestions for improvement.